### PR TITLE
Fix sharing of mouse button state from mousekeys to ps2_mouse

### LINKED ISSUE #1339: Mouse keys messed up for anyone else?
Seems like after [this change](https://github.com/qmk/qmk_firmware/pull/1327/files#diff-eda9a477b1fb87dced86023b342008c4L314), `KC_MS_BTN1` and  `KC_MS_BTN2` stopped working on my boards. Anyone else experience this, or is it just the EZ?

### PR DESCRIPTION
**Pull request by manna-harbour**
Fix sharing of mouse button state from mousekeys to ps2_mouse.

## Description

#1327 shares mousekeys button state with ps2_mouse but prevents clicks when using only mousekeys.

#1344 fixes this but produces clicks from both mousekeys and ps2_mouse.

https://github.com/qmk/qmk_firmware/issues/1339#issuecomment-304580379 would fix the missing and duplicate cilcks but break mousekeys dragging.

With this change, when ps2_mouse is disabled, mousekeys works as usual.  With ps2_mouse enabled, mousekeys button state is shared with ps2_mouse for clicking, dragging, and scrolling, mousekeys clicks are produced by ps2_mouse only, and mousekeys button state is transferred to mousekeys without generating clicks to enable mousekeys dragging. 

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
Resolves #1339.
* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Erovia** [comment]
Don't you need to call `mousekey_send();` for PS2 mousekeys?

**Comment by manna-harbour** [comment]
Thanks for having a look at this.

> Don't you need to call `mousekey_send();` for PS2 mousekeys?

Actually, that was the problem.  With ps2_mouse enabled, action.c and ps2_mouse.c were both sending click events.

Ultimately, it would be nice if mousekeys, ps2_mouse, and pointing_device all had their mouse buttons properly integrated, but just using the code QMK has inherited from TMK and the subsequent attempts to fix it, I believe this is what it should look like.  I've been using this daily without issue.

**Comment by manna-harbour** [comment]
> `__attribute__ ((weak))`
> LGTM

strong thanks!

File: tmk_core/common/action.c [hunk 1]
```diff
@@ -394,36 +394,46 @@ void process_action(keyrecord_t *record, action_t action) {
...
+#    endif
                     default:
+                        mousekey_off(action.key.code);
+                        mousekey_send();
                         break;
                 }
-                mousekey_off(action.key.code);
-                mousekey_send();
             }
             break;
```
Comment (drashna) [inline tmk_core/common/action.c:438]: This is a nitpick, but the way it looks ... bugs me.  Wouldn't this be better: 
```c
        case ACT_MOUSEKEY:
            if (event.pressed) {
                mousekey_on(action.key.code);
                switch (action.key.code) {
#    ifdef PS2_MOUSE_ENABLE
                    case KC_MS_BTN1:
                        tp_buttons |= (1 << 0);
                        break;
                    case KC_MS_BTN2:
                        tp_buttons |= (1 << 1);
                        break;
                    case KC_MS_BTN3:
                        tp_buttons |= (1 << 2);
                        break;
#    endif
                    default:
                        mousekey_send();
                        break;
                }
            } else {
                mousekey_off(action.key.code);
                switch (action.key.code) {
#    ifdef PS2_MOUSE_ENABLE
                    case KC_MS_BTN1:
                        tp_buttons &= ~(1 << 0);
                        break;
                    case KC_MS_BTN2:
                        tp_buttons &= ~(1 << 1);
                        break;
                    case KC_MS_BTN3:
                        tp_buttons &= ~(1 << 2);
                        break;
#    endif
                    default:
                        mousekey_send();
                        break;
                }
            }
            break;
```

Comment (manna-harbour) [reply tmk_core/common/action.c:438]: Thanks, that's much better.  I'll change it.

**Comment by manna-harbour** [comment]
Thanks everyone!

**Pull request merged**